### ISSUE TITLE
Whoops a critical error in domains widget has occured 

### ISSUE THREAD

**Issue by straussmaus** [QubesOS/qubes-issues#6135]
**Qubes OS version**
R4.0

**Affected component(s) or functionality**

On Boot

**Brief summary**

Every time I boot I get this error in a popup window:

"Whoops a critical error in domains widget has occured "

I click ok and it goes away and everything seems to work just fine.

**To Reproduce**

Boot Qubes

**Expected behavior**
I would not expect to see this error, or I would expect to see instructions as to what to do.

**Additional context**
I can provide the full error message if requested.  It's just a lot of typing, and I have to remember on boot to write it down.

**Solutions you've tried**

Checked for upgrades to dom0 (there are none)

Note that this may have started around the most recent dom0 upgrade, but I did not make a clear connection.

**Relevant [documentation](https://www.qubes-os.org/doc/) you've consulted**
None

**Related, [non-duplicate](https://www.qubes-os.org/doc/reporting-bugs/#new-issues-should-not-be-duplicates-of-existing-issues) issues**
https://github.com/QubesOS/qubes-issues/issues/4481 (seems unrelated on detailed look)

**Comment by andrewdavidwong** [comment]
> I can provide the full error message if requested. It's just a lot of typing, and I have to remember on boot to write it down.

Is it not possible to copy/paste the error message from dom0 using the Qubes Clipboard widget?

**Comment by straussmaus** [comment]
> Is it not possible to copy/paste the error message from dom0 using the Qubes Clipboard widget?

No it is not.  The text can not be highlighted on the error bubble (like all Qubes dom0 error bubbles).

Here is the text:

>Whoops.  A critical error in Domains Widge has occured.  this is most likely a bug in th ewidget.  To restart the widget, run 'qui-domains' in dom0.

I run qui-domains in dom0 and there is no reply from the console.

**Comment by marmarta** [comment]
But after you run qui-domains, does the widget appear? 

Also, could you check `journalctl -e` output next time this happens? Perhaps there's an interesting error there.

**Comment by andrewdavidwong** [comment]
> I don't know what "Domains Widget" is supposed to mean if that is what you are asking me.

From https://www.qubes-os.org/doc/getting-started/#gui-and-command-line-tools:

> The **Domains Widget** allows you to manage running qubes, turn them on and off, and monitor memory usage.
[...]
![widgets](https://www.qubes-os.org/attachment/wiki/GettingStarted/r4.1-q40_widgets.png)

(In Qubes 4.0, the icon is the blue Q icon. This screenschot is from 4.1.)

**Comment by straussmaus** [comment]
TY but I still don't know what you mean.  I don't see any change when I run the command in dom0.  I have never seen that cube/Qube icon that you're pointing to.  I see a blue "Q" but that seems fine.

**Comment by andrewdavidwong** [comment]
> I have never seen that cube/Qube icon that you're pointing to. I see a blue "Q" but that seems fine.

As I mentioned, the icon for the Qubes Domains Widget changes from a blue "Q" in 4.0 to a blue cube in 4.1. (Think of this image as a preview of what it will look like in 4.1., but it still the same widget.) Since you are running 4.0 and not 4.1, it is expected that you would have seen only the blue "Q," not the blue cube.

**Comment by hackerncoder** [comment]
This is how it looks in 4.0: 
![r4.0 domains widget](https://www.qubes-os.org/attachment/wiki/GettingStarted/q40_widgets.png)

**Comment by straussmaus** [comment]
Right.  Like I said, "I see a blue 'Q' but that seems fine."  What I mean is that when I boot, I have the blue Q.  I get the error, but the Q is there.  I click on it and I see the domains. 

But now that I am looking closer, I see when I run qui-domains and the console hangs on dom0, a second blue Q shows up that behaves exactly like the first.

**Comment by straussmaus** [comment]
This is happening on a fresh install of 4.0.3.

**Comment by straussmaus** [comment]
This continues to happen on a freshly installed Qubes 4.0.3 where I restored my qubes, but not dom0.  dom0 is the default that came with the Qubes 4.0.3 install.

**Comment by unman** [comment]
On Tue, Nov 24, 2020 at 12:03:16AM -0800, straussmaus wrote:
> This is happening on a fresh install of 4.0.3.
> 

Do you mean on a vanilla fresh install? A clean install with default
qubes?

**Comment by straussmaus** [comment]
Yes to the vanilla and fresh but I did not see the error on first run, then imported qubes and after one or five reboots, I see the error again.  Would I have seen the error if I did not restore the qubes?  I don't know.

**Comment by unman** [comment]
On Thu, Dec 10, 2020 at 08:49:44PM -0800, straussmaus wrote:
> Yes to the vanilla and fresh but I did not see the error on first run, then imported qubes and after one or five reboots, I see the error again.  Would I have seen the error if I did not restore the qubes?  I don't know.
> 

Since this error has not been reported widely, if at all, that seems a
reasonable conclusion.
Given the number of installs *without* this error, I would say it is a
(near) certainty.

**Comment by straussmaus** [comment]
@unman

I think you must mean the opposite of what you wrote?  That I would *not* see the error if I did not restore the qubes.  Right?  That implies that the restore somehow is causing the error.  

Because the alternative (what you said) is the error showing up on a 100% fresh vanilla install that has not been used, and "Given the number of installs *without* this error" that does not make sense.  Yes?

So based on the assumption that the error is caused by my qubes, the restore, or the way they link together, please suggest a course of action to debug.  The error does not indicate its own source, restoring and using qubes should be acceptable use, so this is still a bug/issue to resolve.

**Comment by marmarta** [comment]
Possible steps to debug: after you run `qui-domains` in dom0, check `journalctl -e` in dom0 to see if there's anything with domains or any Python errors there.

**Comment by iamahuman** [comment]
Has marmarek/qubes-desktop-linux-manager@f4ae49d775d63b094c9dd9e2cb1776ee0479aa17 been backported?

**Comment by straussmaus** [comment]
No @marmarta thank you.  This issue started again after I migrated computers in January.  I will try this when it happens again.

**Comment by straussmaus** [comment]
@marmarta 

Okay this happened again.

I still do not understand why I must run qui-domains.  All it does it create another "Q" icon next to the one that is already there.  The items that are revealed on-click on the second Q are different from the first.  The manual one has a more complete list.  Is that important?

But I did this.  Then I ran `journalctl -e` and there are no messages about qui-domians.  The most recent messages are still from a minute or two before, and have to do with sys-whonix.

**Comment by donob4n** [comment]
@straussmaus If you can paste the error it would be very helpful. If you can select it and copy/paste in dom0, paste it to some file and then move to an AppVM with `qvm-move-to-vm destVM logfile` if not possible you could also use a screenshot.

**Comment by straussmaus** [comment]
Okay I will do this next time, but I think what I wrote is the error.  Why would a screenshot help?

**Comment by iamahuman** [comment]
I don't understand why this issue is still open—it is already fixed in QubesOS/qubes-desktop-linux-manager#104 for R4.1.

**Comment by donob4n** [comment]
> Okay I will do this next time, but I think what I wrote is the error. Why would a screenshot help?

Sorry I didn't see you already posted the error. I meant an screenshot of the error message. Maybe the error info should be improved.

**Comment by andrewdavidwong** [comment]
> Correct me if I'm wrong, but  I don't understand why this issue is still open—it is already fixed in QubesOS/qubes-desktop-linux-manager#104 for R4.1.

That PR referenced this issue but did not include a "close" keyword. (@marmarek, was that intentional?) It is also unclear whether fixing this in 4.1 is sufficient or whether this also needs to be fixed in 4.0.

**Comment by straussmaus** [comment]
I'm not sure that https://github.com/QubesOS/qubes-desktop-linux-manager/pull/104 is related.

I'm not sure it isn't.  But the title of https://github.com/QubesOS/qubes-desktop-linux-manager/pull/104 is "Don't crash if VM was quickly added and then removed" and this error has nothing to do with adding or removing a VM/Qube.  It only happens on boot as described in the first post.  It does not happen during a session.

I've provided the error message and run qui-domains and `journalctl -e` and posted about that.

If there is more I can do to give you better information then please tell me and I will.

**Comment by straussmaus** [comment]
This error continues to happen sometimes.  It is not predictable.  It is also less common than it was before.

One hypothesis, could it be related to low disk space?

I do not think it is related to #104.

**Comment by straussmaus** [comment]
This continues to happen on 4.1, with plenty of disk space.  Not often.  Actually it's rare now.  But it happens.

**Comment by DemiMarie** [comment]
@straussmaus Can you provide a backtrace?

**Comment by straussmaus** [comment]
@DemiMarie I'm not sure what you mean by backtrace in this context.  I usually use that word as a function within a compiler.  Can you explain?

**Comment by DemiMarie** [comment]
@straussmaus When the domains widget crashes, it should print a stack trace to stderr.  That will be logged by `systemd-journald`.  The stack trace is what the developers need to figure out what went wrong.

**Comment by straussmaus** [comment]
@DemiMarie So by "backtrace" and by telling me which daemon handles logging, are you asking me to check journalctl next time I get this error?

**Comment by DemiMarie** [comment]
> @DemiMarie So by "backtrace" and by telling me which daemon handles logging, are you asking me to check journalctl next time I get this error? (Not trying to be cute, I'm honestly asking.)

Yes (and I should have phrased this better).

**Comment by straussmaus** [comment]
> Given the number of installs *without* this error, I would say it is a
(near) certainty.

This still happens on 4.10.  I have not checked `journalctl -e` and I know I should, but I never seem to remember when it happens.

Does no one else have this problem yet?
